hackathon_site/event/views.py

```diff
@@ -200,3 +207,38 @@ def post(self, request, *args, **kwargs):
         at once.
         """
         return super().post(request, *args, **kwargs)
+
+
+class LeaveTeamView(generics.GenericAPIView):
+    permission_classes = [UserHasProfile]
+
+    @transaction.atomic
+    @swagger_auto_schema(responses={201: TeamSerializer})
+    def post(self, request, *args, **kwargs):
+        profile = request.user.profile
+        team = profile.team
+
+        # Raise 400 if team has active orders
+        active_orders = OrderItem.objects.filter(
+            Q(part_returned_health__isnull=True),
+            ~Q(order__status="Cancelled"),
+            Q(order__team=team),
+        )
+        if active_orders.exists():
+            return Response(
```

Comment:
Since this is still an API view, can you craft this so it's the same shape as a normal DRF 400 response? That will let us use some helpers on the frontend to display the error with less edge cases.

You may be able to throw a DRF validation error to accomplish that.